src/assets/scripts/peer.js

```diff
@@ -0,0 +1,90 @@
+import SimplePeer from 'simple-peer';
+import Debug from 'debug';
+
+const peerStore = [];
+
+function signalPeer(peerId, signalData) {
+  const peer = peerStore.find(peer => peer.id === peerId);
+  if (!peer) {
+    throw new Error(`Peer "${peerId}" not found`);
+  }
+  peer.signal(signalData);
+}
+
+class Deferred {
+  constructor() {
+    this.promise = new Promise((resolve, reject) => {
+      this.resolve = resolve;
+      this.reject = reject;
+    });
+  }
+
+  then() {
+    return this.promise.then(...arguments);
+  }
+
+  catch() {
+    return this.promise.catch(...arguments);
+  }
+}
+
+export class Peer {
+  constructor(client, id, room, initiate) {
+    this.client = client;
+    this.id = id;
+    this.room = room;
+    this.debug = Debug(`app:peer:${this.id}`);
+    this._receivers = [];
+    this.connect = new Deferred();
+    this.connection = new SimplePeer({initiator: initiate});
```

Comment:
`connection` is pretty generic,  I would think its a websocket connection before I think its a WebRTC data channel, perhaps `dataChannel` is a better name.